napari/_qt/qt_event_loop.py

```diff
@@ -127,6 +128,13 @@ def get_app(
         app.setWindowIcon(QIcon(kwargs.get('icon')))
         set_app_id(kwargs.get('app_id'))
 
+        if os.getenv("NAPARI_CATCH_ERRORS") not in ('0', 'False'):
+            from napari.notifications import notification_manager
+
+            notification_manager.notification_ready.connect(
+                NapariNotification.show_notification
+            )
```

Comment:
This is actually going down to here; if I change NapariNotification.show_notification to a local function and add `gc.collect()` in here; it has the same effect of crashing napari; with NapariNotification object not staying around. being reclaimed.

I'm going to try to change that to a bound method on notification_manager, which can keep a reference to the  notification.